internal/provider/waypoint/resource_waypoint_application.go

```diff
@@ -0,0 +1,426 @@
+	ReadmeMarkdown          types.String `tfsdk:"readme_markdown"`
+	ApplicationTemplateID   types.String `tfsdk:"application_template_id"`
+	ApplicationTemplateName types.String `tfsdk:"application_template_name"`
+	NamespaceID             types.String `tfsdk:"namespace_id"`
+
+	// deferred for now
+	// Tags       types.List `tfsdk:"tags"`
+
+	// deferred and probably a list or objects, but may possible be a separate
+	// ActionCfgs types.List `tfsdk:"action_cfgs"`
+}
+
+func (r *ApplicationResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_waypoint_application"
+}
+
+func (r *ApplicationResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "The Waypoint Application resource managed the lifecycle of an Application that's based off of a Template.",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:    true,
+				Description: "The ID of the Application.",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"name": schema.StringAttribute{
+				Description: "The name of the Application.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"organization_id": schema.StringAttribute{
+				Description: "The ID of the HCP organization where the Waypoint Application is located.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"project_id": schema.StringAttribute{
+				Description: "The ID of the HCP project where the Waypoint Application is located.",
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"application_template_id": schema.StringAttribute{
+				Required:    true,
+				Description: "ID of the Application Template this Application is based on.",
+			},
+			// application_template_name is a computed only attribute for ease
+			// of reference
+			"application_template_name": schema.StringAttribute{
+				Computed:    true,
+				Description: "Name of the Application Template this Application is based on.",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"readme_markdown": schema.StringAttribute{
+				Computed: true,
+				Optional: true,
+				Description: "Instructions for using the Application (markdown" +
+					"format supported). Note: this is a base64 encoded string, and" +
+					"can only be set in configuration after initial creation. The" +
+					"initial version of the README is generated from the README" +
```

Comment:
```suggestion
					" initial version of the README is generated from the README " +
```
Also should be spaces here